cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -96,6 +96,17 @@ bool ends_with_utf8_utf8(const char* data, gdv_int32 data_len, const char* suffi
           (memcmp(data + data_len - suffix_len, suffix, suffix_len) == 0));
 }
 
+FORCE_INLINE
+bool is_substr_utf8_utf8(const char* data, int32_t data_len, const char* substr,
+                         int32_t substr_len) {
+  for (int32_t i = 0; i <= data_len - substr_len; ++i) {
+    if (memcmp(data + i, substr, substr_len) == 0) {
```

Comment:
You might open a JIRA issue about comparing this (O(m * n)) with an implementation of the KMP algorithm (O(m + n) https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm). For small m and n it may not be worth the trouble

Reply:
There is already an open ticket to use a linear time algorithm like Boyer Moore for string matching (https://issues.apache.org/jira/browse/ARROW-7278). I will run some micro-benchmarks to compare this patch with a KMP/BM implementation.